pymc3/distributions/bart.py

```diff
@@ -167,6 +169,19 @@ def draw_leaf_value(self, idx_data_points):
         draw = self.mean(R_j)
         return draw
 
+    def predict(self, X_new):
+        """Compute out of sample predictions evaluated at X_new"""
+        trees = self.all_trees
+        num_observations = X_new.shape[0]
+        pred = np.zeros((len(trees), num_observations))
+        np.random.randint(len(trees))
+        for draw, trees_to_sum in enumerate(trees):
+            new_Y = np.zeros(X_new.shape[0])
+            for tree in trees_to_sum:
+                new_Y += [tree.predict_out_of_sample(x) for x in X_new]
+            pred[draw] = new_Y
+        return pred
```

Comment:
am i correct in my understanding that this will return a range of predictions for each `X_new` observation, that you could then use to get a mean estimate + credible interval?

Reply:
right, this will return the mean function predictions